### PR TITLE
Run tests against DynamoDB.

### PR DESCRIPTION
**Pull request by samiwel**
### What is the context of this PR?
Until now the feature branch has been running tests against the filesystem. The preferred way of running the tests are against DynamoDB since this is what will be used initially within AWS.

This change spins up a containerised DynamoDB local instance when running the tests and makes slight changes to the application in order to get the tests passing.

### How to test
The API tests should pass consistently when running `yarn test`.
Should observe that the tests are running against DynamoDB.

### CODE REVIEW

**Comment by tgandrews** [comment]
Looks like it needs to be rebased against 797

**Comment by samiwel** [comment]
@tgandrews Yeah, I started a rebase but the merges looked pretty horrible. I'm wondering if it might be easier to cherry pick some of the later commits on top of 797 instead.

**Comment by tgandrews** [comment]
yeah, the hack required for the dynamoose model I don't think is needed anymore as the filesystem still has a difference between the persisted format and the graphql format. So it grabs the one from the context so the model/questionnaire stuff can probably be thrown away.

**Comment by samiwel** [comment]
[Fix API tests for JSON backend store](https://trello.com/c/RvpS8d1f/913-fix-api-tests-for-json-backend-store)

File: eq-author-api/utils/datastoreDynamo.js [hunk 1]
```diff
@@ -44,13 +43,21 @@ const getQuestionnaire = id => {
...
+          questionnaire.metadata = [];
+        }
         resolve(questionnaire);
       });
   });
 };
 
-const MAX_UPDATE_TIMES = 2;
-const saveQuestionnaire = async (questionnaire, count = 0) => {
+const MAX_UPDATE_TIMES = 1;
```
Comment (tgandrews) [inline eq-author-api/utils/datastoreDynamo.js]: We can increase this for cypress back to `2`

File: eq-author-api/schema/tests/page.test.js [hunk 2]
```diff
@@ -47,7 +51,7 @@ describe("page", () => {
       expect(createdPage).toMatchObject({
         title: "Title",
         description: "Description",
-        position: 0,
+        position: 1,
```
Comment (tgandrews) [inline eq-author-api/schema/tests/page.test.js]: This and the above change I don't think are necessary.

File: eq-author-api/package.json [hunk 3]
```diff
@@ -43,7 +43,7 @@
     "pg-hstore": "latest",
     "pino-noir": "latest",
     "uuid": "latest",
-    "wait-for-postgres": "latest"
+    "wait-on": "^3.2.0"
```
Comment (tgandrews) [inline eq-author-api/package.json]: Can we point this on `latest`? 

It should just be a case of updating this and running `yarn`.

File: eq-author-api/schema/resolvers/base.js [hunk 4]
```diff
@@ -669,7 +674,8 @@ const Resolvers = {
     introduction: section => (section.introductionEnabled ? section : null),
     availablePipingAnswers: ({ id }, args, ctx) =>
       getPreviousAnswersForSection(ctx.questionnaire, id),
-    availablePipingMetadata: (section, args, ctx) => ctx.questionnaire.metadata,
+    availablePipingMetadata: (section, args, ctx) =>
+      ctx.questionnaire.metadata || [],
```
Comment (tgandrews) [inline eq-author-api/schema/resolvers/base.js]: I don't think this is needed now the change in `getQuestionnaire` is in.

File: eq-author-api/schema/resolvers/base.js [hunk 5]
```diff
@@ -696,7 +702,8 @@ const Resolvers = {
     confirmation: page => page.confirmation,
     availablePipingAnswers: ({ id }, args, ctx) =>
       getPreviousAnswersForPage(ctx.questionnaire, id),
-    availablePipingMetadata: (page, args, ctx) => ctx.questionnaire.metadata,
+    availablePipingMetadata: (page, args, ctx) =>
+      ctx.questionnaire.metadata || [],
```
Comment (tgandrews) [inline eq-author-api/schema/resolvers/base.js]: As above

File: eq-author-api/schema/resolvers/base.js [hunk 6]
```diff
@@ -964,7 +971,8 @@ const Resolvers = {
     },
     availablePipingAnswers: ({ id }, args, ctx) =>
       getPreviousAnswersForPage(ctx.questionnaire, id),
-    availablePipingMetadata: (page, args, ctx) => ctx.questionnaire.metadata,
+    availablePipingMetadata: (page, args, ctx) =>
+      ctx.questionnaire.metadata || [],
```
Comment (tgandrews) [inline eq-author-api/schema/resolvers/base.js]: As above

**Pull request merged**